host/xtest/aes_gcm_counter_overflow_test_decrypt.h

```diff
@@ -0,0 +1,653 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright 2024 NXP
+ */
+
+{
+	.algo = TEE_ALG_AES_GCM, .mode = TEE_MODE_DECRYPT, .key_type = TEE_TYPE_AES,
+	.key = (const uint8_t []){ 0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xaa, 0xbb, 0xcc, 0xdd, 0xee, 0xff, },
+	.key_len = 16,
+	.nonce = (const uint8_t []){ 0xf9, 0x5f, 0xde, 0x4a, 0x75, 0x19, 0x13, 0x20, 0x2a, 0xee, 0xee, 0x32, 0xa0, 0xb5, 0x57, 0x53, },
+	.nonce_len = 16,
+	.aad_incr = 0,
+	.aad = NULL,
+	.aad_len = 0,
+	.in_incr = 0,
+	.ptx = (const uint8_t []){ 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, },
+	.ptx_len = 40,
+	.ctx = (const uint8_t []){ 0x00, 0x07, 0x8d, 0x10, 0x9d, 0x92, 0x14, 0x3f, 0xcd, 0x5d, 0xf5, 0x67, 0x21, 0xb8, 0x84, 0xfa, 0xc6, 0x4a, 0xc7, 0x76, 0x2c, 0xc0, 0x9e, 0xea, 0x2a, 0x3c, 0x68, 0xe9, 0x2a, 0x17, 0xbd, 0xb5, 0x75, 0xf8, 0x7b, 0xda, 0x18, 0xbe, 0x56, 0x4e, },
+	.ctx_len = 40,
+	.tag = (const uint8_t []){ 0x15, 0x2a, 0x65, 0x04, 0x5f, 0xe6, 0x74, 0xf9, 0x76, 0x27, 0x42, 0x7a, 0xf5, 0xbe, 0x22, 0xda, },
+	.tag_len = 16,
+	.line = 77
```

Comment:
What does this line number refers to?
Looking at the vectors generated from NIST GCM vectors, the line number refers to the line in the deflated .rsp file. Here it corresponds to the vector files you process but these files are not reacheable here.

Reply:
I think it's worth mentioning that in the commit message that the line number refers the `"tcId"` in the reference Json file. This  number is printed by xtest when a test fails to help one finding the relted vector.

Reply:
> Sorry for the noise but since you use a nice script to produce these arrays [...]

Should that script be committed too? It would help in case one wants to update/extend the data.

Reply:
> Should that script be committed too? It would help in case one wants to update/extend the data.

yes, it could be added to the scripts folder if it is ok.

Reply:
> > Should that script be committed too? It would help in case one wants to update/extend the data.
> 
> yes, it could be added to the scripts folder, but it currently only support AES GCM test vectors from google/wycheproof.

That's OK, you can mention it in the script as a comment.

Reply:
> it corresponds to the test id: https://github.com/google/wycheproof/blob/master/testvectors_v1/aes_gcm_test.json#L1141
> 
> to find easily which test vector pass or not.

I find this a bit confusing. We're quite used to the line number being the line number in the C code. Perhaps we should have another field (`.extra_id`?) which would be reported in addition to the line number when non-zero for instance? 

```
o regression_4005.43 AE case 42 algo 0x40000810 line 2609
  regression_4005.43 OK
o regression_4005.44 AE case 43 algo 0x40000810 line 2609
  regression_4005.44 OK
o regression_4005.45 AE case 44 algo 0x40000810 line 22 extra_id 77
  regression_4005.45 OK
o regression_4005.46 AE case 45 algo 0x40000810 line 40 extra_id 78
  regression_4005.46 OK
```

Reply:
something like that 86b73e5c7b8e6b31c6c5ac496ad8b96d7bd2e579 ?